### ISSUE TITLE
SiUsbXpress driver install error

### ISSUE THREAD

**Issue by JustTheDazz** [EvanMulawski/FanControl.CorsairLink#185]
When I go to right click on the install the driver for corsair(right click on CorsairSiUSBXp) I get this error message:
![corsairthing](https://github.com/user-attachments/assets/0885be2b-d7f6-47b2-9c6c-7756001a857e)

**Comment by EvanMulawski** [comment]
Hi @JustTheDazz, please try the one in a v1.6.x release. This issue may be caused by some automations being used in v1.7.0. Thanks!

**Comment by SerinitySW** [comment]
I also experienced this issue, and can confirm the 1.6.x driver successfully installs. However, the plugin still does not function:

```
2024-08-19T17:31:22.4294633Z [INF] Plugin: Version: 1.7.0-beta.1+af7a336a05f6df1755e8629377528c8739f9d14d
2024-08-19T17:31:22.4519277Z [INF] HID Device Manager: Found 0 Corsair HID device(s)

2024-08-19T17:31:22.4543629Z [INF] HID Device Manager: Found 0 supported Corsair HID device(s)

2024-08-19T17:31:22.4611733Z [WRN] Device Enumeration: Failed to enumerate SiUsbXpress devices. This can be ignored if no devices require this driver.
2024-08-19T17:31:24.5261067Z [INF] Plugin: Version: 1.7.0-beta.1+af7a336a05f6df1755e8629377528c8739f9d14d
2024-08-19T17:31:24.5262291Z [INF] HID Device Manager: Found 0 Corsair HID device(s)
```

...

```
2024-08-19T21:27:38.6859026Z [WRN] Device Enumeration: Failed to enumerate SiUsbXpress devices. This can be ignored if no devices require this driver.
2024-08-19T21:27:38.6908638Z [DBG] Device Enumeration: 
Type: System.DllNotFoundException
Message: Unable to load DLL 'SiUSBXp.dll' or one of its dependencies: The specified module could not be found. (0x8007007E)
Source: FanControl.CorsairLink.merged
HResult: -2146233052
Stack Trace:
   at CorsairLink.SiUsbXpress.Driver.SiUsbXpressDriver.SI_GetNumDevices(UInt32& lpdwNumDevices)
   at CorsairLink.SiUsbXpress.Driver.SiUsbXpressDriverHelper.EnumerateDevices()
   at CorsairLink.SiUsbXpress.Driver.SiUsbXpressDeviceEnumerator.Enumerate()
   at CorsairLink.SiUsbXpressDeviceManager.GetSupportedDevices(IDeviceGuardManager deviceGuardManager, ILogger logger)
   at FanControl.CorsairLink.CorsairLinkPlugin.GetSupportedDevices()

2024-08-19T21:27:40.9246540Z [INF] Plugin: Version: 1.7.0-beta.2+874384851d3f695a4aed1bbf2ea9a918962dc7dd
2024-08-19T21:27:40.9249335Z [INF] HID Device Manager: Found 0 Corsair HID device(s)

2024-08-19T21:27:40.9249562Z [INF] HID Device Manager: Found 0 supported Corsair HID device(s)

2024-08-19T21:27:40.9274421Z [WRN] Device Enumeration: Failed to enumerate SiUsbXpress devices. This can be ignored if no devices require this driver.
2024-08-19T21:27:40.9275942Z [DBG] Device Enumeration: 
Type: System.DllNotFoundException
Message: Unable to load DLL 'SiUSBXp.dll' or one of its dependencies: The specified module could not be found. (0x8007007E)
Source: FanControl.CorsairLink.merged
HResult: -2146233052
Stack Trace:
   at CorsairLink.SiUsbXpress.Driver.SiUsbXpressDriver.SI_GetNumDevices(UInt32& lpdwNumDevices)
   at CorsairLink.SiUsbXpress.Driver.SiUsbXpressDriverHelper.EnumerateDevices()
   at CorsairLink.SiUsbXpress.Driver.SiUsbXpressDeviceEnumerator.Enumerate()
   at CorsairLink.SiUsbXpressDeviceManager.GetSupportedDevices(IDeviceGuardManager deviceGuardManager, ILogger logger)
   at FanControl.CorsairLink.CorsairLinkPlugin.GetSupportedDevices()

2024-08-19T21:27:41.8878182Z [INF] Plugin: Version: 1.7.0-beta.2+874384851d3f695a4aed1bbf2ea9a918962dc7dd
2024-08-19T21:27:41.8879080Z [INF] HID Device Manager: Found 0 Corsair HID device(s)

2024-08-19T21:27:41.8879242Z [INF] HID Device Manager: Found 0 supported Corsair HID device(s)

2024-08-19T21:27:41.8903638Z [WRN] Device Enumeration: Failed to enumerate SiUsbXpress devices. This can be ignored if no devices require this driver.
2024-08-19T21:27:41.8905135Z [DBG] Device Enumeration: 
Type: System.DllNotFoundException
Message: Unable to load DLL 'SiUSBXp.dll' or one of its dependencies: The specified module could not be found. (0x8007007E)
Source: FanControl.CorsairLink.merged
HResult: -2146233052
Stack Trace:
   at CorsairLink.SiUsbXpress.Driver.SiUsbXpressDriver.SI_GetNumDevices(UInt32& lpdwNumDevices)
   at CorsairLink.SiUsbXpress.Driver.SiUsbXpressDriverHelper.EnumerateDevices()
   at CorsairLink.SiUsbXpress.Driver.SiUsbXpressDeviceEnumerator.Enumerate()
   at CorsairLink.SiUsbXpressDeviceManager.GetSupportedDevices(IDeviceGuardManager deviceGuardManager, ILogger logger)
   at FanControl.CorsairLink.CorsairLinkPlugin.GetSupportedDevices()
```

The above is what is logged in CorsairLink.log when debug is turned on. Personally I have an [h115i](https://www.corsair.com/us/en/p/refurb-cpu-coolers/cw-9060027-ww-rf/cw-9060027-ww-rf), which seems to be supported from what the readme says. For some reason the driver just isn't getting picked up. 

Windows 10 22H2
FanControl Release [v199](https://github.com/Rem0o/FanControl.Releases/releases/tag/V199)
FanControl.CorsairLink [v1.7.0-beta.2](https://github.com/EvanMulawski/FanControl.CorsairLink/releases/tag/v1.7.0-beta.2)
HWiNFO v8.06-5500

Ryzen 7 1700
ASUS Prime X370-A
Corsair H115i (CW-9060027-WW-RF)

**Comment by EvanMulawski** [comment]
Hi @SerinitySW, please make sure the SiUsbXp.dll file is copied to the Plugins directory and is unblocked.

**Comment by EvanMulawski** [comment]
Resolved in [v1.7.0-beta.3](https://github.com/EvanMulawski/FanControl.CorsairLink/releases/tag/v1.7.0-beta.3)
